### ISSUE TITLE
F821 false positive: `dict.__dict__['fromkeys']`

### ISSUE THREAD

**Issue by AA-Turner** [astral-sh/ruff#9307]
Ruff doesn't like classmethod descriptors, or seems to think that they're undefined, which is rather odd.

```python
a = type({}).__dict__['fromkeys']
b = dict.__dict__['fromkeys']
assert a is b
```

Saving the above to `bug.py`, and running only with `F821`:

```pscon
PS I:\Development> ruff -V
ruff 0.1.9
PS I:\Development> ruff --isolated --show-source --select F821 bug.py
bug.py:2:20: F821 Undefined name `fromkeys`
  |
1 | a = type({}).__dict__['fromkeys']
2 | b = dict.__dict__['fromkeys']
  |                    ^^^^^^^^ F821
3 | assert a is b
  |

Found 1 error.
```

Note only the second line yields an error -- using ``type({})`` as an indirection for ``dict`` solves the issue.

A

**Comment by charliermarsh** [comment]
Thanks — I’m guessing it thinks it’s a type annotation, as when subscripting dict directly.

**Comment by charliermarsh** [comment]
Closed by https://github.com/astral-sh/ruff/pull/9309.

**Comment by AA-Turner** [comment]
Thanks Charlie!

A
